code/game/mecha/equipment/tools/tools.dm

```diff
@@ -827,6 +826,88 @@
 				A.use_power(delta*ER.coeff, pow_chan)
 	return
 
+/obj/item/mecha_parts/mecha_equipment/combat_shield
+	name = "linear combat shield"
+	desc = "A shield generator that forms a rectangular, unidirectionally projectile-blocking wall in front of the exosuit."
+	icon_state = "shield"
+	origin_tech = list(TECH_PHORON = 3, TECH_MAGNET = 6, TECH_ILLEGAL = 4)
+	equip_cooldown = 5
+	energy_drain = 20
+	range = 0
+
+	var/obj/item/shield_projector/line/exosuit/my_shield = null
+	var/my_shield_type = /obj/item/shield_projector/line/exosuit
+	var/icon/drone_overlay
+
+	equip_type = EQUIP_HULL
+
+/obj/item/mecha_parts/mecha_equipment/combat_shield/New()
+	..()
+	my_shield = new my_shield_type
+	my_shield.shield_regen_delay = equip_cooldown
+	my_shield.my_tool = src
+	return
+
+/obj/item/mecha_parts/mecha_equipment/combat_shield/critfail()
+	..()
+	my_shield.adjust_health(-200)
+	return
+
+/obj/item/mecha_parts/mecha_equipment/combat_shield/Destroy()
+	chassis.overlays -= drone_overlay
+	my_shield.forceMove(src)
+	my_shield.destroy_shields()
+	my_shield.my_tool = null
+	my_shield.my_mecha = null
+	qdel(my_shield)
+	my_shield = null
+	..()
+
+/obj/item/mecha_parts/mecha_equipment/combat_shield/attach(obj/mecha/M as obj)
+	..()
+	if(chassis)
+		my_shield.shield_health = my_shield.shield_health / 2

```

Comment:
What specifically does this line do?

Reply:
Sets the shield to half health when the shield is attached, as a 'cheese' prevention, and as a windup. I realized writing this that it's equally cheezable.